go/arrow/internal/arrdata/ioutil.go

```diff
@@ -172,6 +173,47 @@ func WriteFile(t *testing.T, f *os.File, mem memory.Allocator, schema *arrow.Sch
 	}
 }
 
+// WriteFile writes a list of records to the given file descriptor, as an ARROW file.
+func WriteFileCompressed(t *testing.T, f *os.File, mem memory.Allocator, schema *arrow.Schema, recs []array.Record, codec flatbuf.CompressionType) {
```

Comment:
at least in other languages I'm familiar with there is some effort to hide the flat buffer details from consumers of the API.  Would it make sense to abstract that here?

Reply:
So this is an internal package (note the "internal" in the path) which means it cannot be imported or called outside of the `arrow` repo here. This is only used for the unit tests so I figured it wasn't a problem to have it use the enum from the flatbuffer details directly in this situation since it's not an externally exposed API for consumers.

The only externally exposed API for consumers is the `WithLZ4` and `WithZstd` functions that were added as options to be passed to the writer which explicitly doesn't expose the flatbuffer details to external consumers.